src/app/lib/device/airplay.js

```diff
@@ -1,56 +1,124 @@
 (function (App) {
     'use strict';
 
-    var airplayer = require('airplayer'),
-        netw = require('network-address'),
-        list = airplayer(),
+    /*
+      Useful AirPlay protocol references:
+      - "Unofficial AirPlay Protocol Specification" (https://nto.github.io/AirPlay.html)
+      - @watson's "airplay-protocol" (https://github.com/watson/airplay-protocol)
+    */
+    var Bonjour = require('bonjour'),
+        AirPlayProtocol = require('airplay-protocol');
+
+    var bonjourBrowser = new Bonjour(),
         collection = App.Device.Collection;
 
-    var makeID = function (baseID) {
-        return 'airplay-' + baseID.replace('.', '');
+    var makeID = function (bonjourService) {
+        // strong de-duplication ID since there can only be 1 bonjour service per host+port
+        var uniqueDeviceId = bonjourService.host + '_' + bonjourService.port;
+        return 'airplay-' + Common.md5(uniqueDeviceId);
```

Comment:
chromecast is probably broken too. there is no point in using md5 there.
add to that device id all you need to make it deduplicatable and debuggeable, but keep it a human readable string, because shady numbers that have no justification make no sense.